src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -11,55 +11,38 @@
 
 namespace Microsoft.DotNet.Watcher.Tools
 {
-    internal sealed class BlazorWebAssemblyDeltaApplier : SingleProcessDeltaApplier
+    internal sealed class BlazorWebAssemblyDeltaApplier(IReporter reporter, BrowserRefreshServer browserRefreshServer) : SingleProcessDeltaApplier
     {
         private const string DefaultCapabilities60 = "Baseline";
         private const string DefaultCapabilities70 = "Baseline AddMethodToExistingType AddStaticFieldToExistingType NewTypeDefinition ChangeCustomAttributes";
         private const string DefaultCapabilities80 = "Baseline AddMethodToExistingType AddStaticFieldToExistingType NewTypeDefinition ChangeCustomAttributes AddInstanceFieldToExistingType GenericAddMethodToExistingType GenericUpdateMethod UpdateParameters GenericAddFieldToExistingType";
 
         private static Task<ImmutableArray<string>>? s_cachedCapabilties;
-        private readonly IReporter _reporter;
         private Version? _targetFrameworkVersion;
         private int _sequenceId;
 
-        public BlazorWebAssemblyDeltaApplier(IReporter reporter)
+        public override void Initialize(ProjectInfo project, string namedPipeName, CancellationToken cancellationToken)
         {
-            _reporter = reporter;
+            base.Initialize(project, namedPipeName, cancellationToken);
+            _targetFrameworkVersion = project.TargetFrameworkVersion;
         }
 
-        public override void Initialize(DotNetWatchContext context, CancellationToken cancellationToken)
-        {
-            Debug.Assert(context.ProcessSpec != null);
-
-            base.Initialize(context, cancellationToken);
-
-            // Configure the app for EnC
-            context.ProcessSpec.EnvironmentVariables["DOTNET_MODIFIABLE_ASSEMBLIES"] = "debug";
-
-            _targetFrameworkVersion = context.FileSet?.Project?.TargetFrameworkVersion;
-        }
-
-        public override Task<ImmutableArray<string>> GetApplyUpdateCapabilitiesAsync(DotNetWatchContext context, CancellationToken cancellationToken)
+        public override Task<ImmutableArray<string>> GetApplyUpdateCapabilitiesAsync(CancellationToken cancellationToken)
         {
             return s_cachedCapabilties ??= GetApplyUpdateCapabilitiesCoreAsync();
 
             async Task<ImmutableArray<string>> GetApplyUpdateCapabilitiesCoreAsync()
             {
-                if (context.BrowserRefreshServer is null)
-                {
-                    throw new ApplicationException("The browser refresh server is unavailable.");
-                }
-
-                _reporter.Verbose("Connecting to the browser.");
+                reporter.Verbose("Connecting to the browser.");
```

Comment:
Do you need a browserRefreshServer null check here or if not, do a Debug.Assert check?
```
            if (browserRefreshServer is null)
            {
                throw new ApplicationException("The browser refresh server is unavailable.");
            }
```

Reply:
No need for neither. `BrowserRefreshServer browserRefreshServer` parameter is declared as not-null (nullability checks are enabled in the entire project).